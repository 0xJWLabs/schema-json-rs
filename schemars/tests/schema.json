{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "schemars__schema__Schema",
  "anyOf": [
    {
      "properties": {
        "Bool": {
          "type": "boolean"
        }
      }
    },
    {
      "properties": {
        "Ref": {
          "$ref": "#/definitions/schemars__schema__SchemaRef"
        }
      }
    },
    {
      "properties": {
        "Object": {
          "$ref": "#/definitions/schemars__schema__SchemaObject"
        }
      }
    }
  ],
  "definitions": {
    "schemars__schema__InstanceType": {
      "enum": [
        "null",
        "boolean",
        "object",
        "array",
        "number",
        "string",
        "integer"
      ]
    },
    "schemars__schema__Schema": {
      "anyOf": [
        {
          "properties": {
            "Bool": {
              "type": "boolean"
            }
          }
        },
        {
          "properties": {
            "Ref": {
              "$ref": "#/definitions/schemars__schema__SchemaRef"
            }
          }
        },
        {
          "properties": {
            "Object": {
              "$ref": "#/definitions/schemars__schema__SchemaObject"
            }
          }
        }
      ]
    },
    "schemars__schema__SchemaObject": {
      "properties": {
        "$id": {
          "anyOf": [
            {
              "type": "string"
            },
            {
              "type": "null"
            }
          ]
        },
        "$schema": {
          "anyOf": [
            {
              "type": "string"
            },
            {
              "type": "null"
            }
          ]
        },
        "allOf": {
          "anyOf": [
            {
              "type": "array",
              "items": {
                "$ref": "#/definitions/schemars__schema__Schema"
              }
            },
            {
              "type": "null"
            }
          ]
        },
        "anyOf": {
          "anyOf": [
            {
              "type": "array",
              "items": {
                "$ref": "#/definitions/schemars__schema__Schema"
              }
            },
            {
              "type": "null"
            }
          ]
        },
        "definitions": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/schemars__schema__Schema"
          }
        },
        "description": {
          "anyOf": [
            {
              "type": "string"
            },
            {
              "type": "null"
            }
          ]
        },
        "enum": {
          "anyOf": [
            {
              "type": "array",
              "items": true
            },
            {
              "type": "null"
            }
          ]
        },
        "extensions": {
          "type": "object",
          "additionalProperties": true
        },
        "items": {
          "anyOf": [
            {
              "$ref": "#/definitions/schemars__schema__SingleOrVec_schemars__schema__Schema_"
            },
            {
              "type": "null"
            }
          ]
        },
        "not": {
          "anyOf": [
            {
              "$ref": "#/definitions/schemars__schema__Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "oneOf": {
          "anyOf": [
            {
              "type": "array",
              "items": {
                "$ref": "#/definitions/schemars__schema__Schema"
              }
            },
            {
              "type": "null"
            }
          ]
        },
        "properties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/schemars__schema__Schema"
          }
        },
        "required": {
          "anyOf": [
            {
              "type": "array",
              "items": {
                "type": "string"
              }
            },
            {
              "type": "null"
            }
          ]
        },
        "title": {
          "anyOf": [
            {
              "type": "string"
            },
            {
              "type": "null"
            }
          ]
        },
        "type": {
          "anyOf": [
            {
              "$ref": "#/definitions/schemars__schema__SingleOrVec_schemars__schema__InstanceType_"
            },
            {
              "type": "null"
            }
          ]
        }
      }
    },
    "schemars__schema__SchemaRef": {
      "properties": {
        "$ref": {
          "type": "string"
        }
      }
    },
    "schemars__schema__SingleOrVec_schemars__schema__InstanceType_": {
      "anyOf": [
        {
          "properties": {
            "Single": {
              "$ref": "#/definitions/schemars__schema__InstanceType"
            }
          }
        },
        {
          "properties": {
            "Vec": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/schemars__schema__InstanceType"
              }
            }
          }
        }
      ]
    },
    "schemars__schema__SingleOrVec_schemars__schema__Schema_": {
      "anyOf": [
        {
          "properties": {
            "Single": {
              "$ref": "#/definitions/schemars__schema__Schema"
            }
          }
        },
        {
          "properties": {
            "Vec": {
              "type": "array",
              "items": {
                "$ref": "#/definitions/schemars__schema__Schema"
              }
            }
          }
        }
      ]
    }
  }
}