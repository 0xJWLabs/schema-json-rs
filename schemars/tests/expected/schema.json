{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "RootSchema",
  "type": "object",
  "properties": {
    "$id": {
      "type": [
        "string",
        "null"
      ]
    },
    "$ref": {
      "type": [
        "string",
        "null"
      ]
    },
    "$schema": {
      "type": [
        "string",
        "null"
      ]
    },
    "additionalItems": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "additionalProperties": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "allOf": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "$ref": "#/definitions/Schema"
      }
    },
    "anyOf": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "$ref": "#/definitions/Schema"
      }
    },
    "const": true,
    "contains": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "default": true,
    "definitions": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/definitions/Schema"
      }
    },
    "deprecated": {
      "type": "boolean"
    },
    "description": {
      "type": [
        "string",
        "null"
      ]
    },
    "else": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "enum": {
      "type": [
        "array",
        "null"
      ],
      "items": true
    },
    "exclusiveMaximum": {
      "type": [
        "number",
        "null"
      ],
      "format": "double"
    },
    "exclusiveMinimum": {
      "type": [
        "number",
        "null"
      ],
      "format": "double"
    },
    "format": {
      "type": [
        "string",
        "null"
      ]
    },
    "if": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "items": {
      "anyOf": [
        {
          "$ref": "#/definitions/SingleOrVec_for_Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "maxItems": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "maxLength": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "maxProperties": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "maximum": {
      "type": [
        "number",
        "null"
      ],
      "format": "double"
    },
    "minItems": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "minLength": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "minProperties": {
      "type": [
        "integer",
        "null"
      ],
      "format": "uint32"
    },
    "minimum": {
      "type": [
        "number",
        "null"
      ],
      "format": "double"
    },
    "multipleOf": {
      "type": [
        "number",
        "null"
      ],
      "format": "double"
    },
    "not": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "oneOf": {
      "type": [
        "array",
        "null"
      ],
      "items": {
        "$ref": "#/definitions/Schema"
      }
    },
    "pattern": {
      "type": [
        "string",
        "null"
      ]
    },
    "patternProperties": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/definitions/Schema"
      }
    },
    "properties": {
      "type": "object",
      "additionalProperties": {
        "$ref": "#/definitions/Schema"
      }
    },
    "propertyNames": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "readOnly": {
      "type": "boolean"
    },
    "required": {
      "type": "array",
      "items": {
        "type": "string"
      }
    },
    "then": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "null"
        }
      ]
    },
    "title": {
      "type": [
        "string",
        "null"
      ]
    },
    "type": {
      "anyOf": [
        {
          "$ref": "#/definitions/SingleOrVec_for_InstanceType"
        },
        {
          "type": "null"
        }
      ]
    },
    "uniqueItems": {
      "type": [
        "boolean",
        "null"
      ]
    },
    "writeOnly": {
      "type": "boolean"
    }
  },
  "additionalProperties": true,
  "definitions": {
    "InstanceType": {
      "enum": [
        "null",
        "boolean",
        "object",
        "array",
        "number",
        "string",
        "integer"
      ]
    },
    "Schema": {
      "anyOf": [
        {
          "type": "boolean"
        },
        {
          "$ref": "#/definitions/SchemaObject"
        }
      ]
    },
    "SchemaObject": {
      "type": "object",
      "properties": {
        "$id": {
          "type": [
            "string",
            "null"
          ]
        },
        "$ref": {
          "type": [
            "string",
            "null"
          ]
        },
        "additionalItems": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "additionalProperties": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "allOf": {
          "type": [
            "array",
            "null"
          ],
          "items": {
            "$ref": "#/definitions/Schema"
          }
        },
        "anyOf": {
          "type": [
            "array",
            "null"
          ],
          "items": {
            "$ref": "#/definitions/Schema"
          }
        },
        "const": true,
        "contains": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "default": true,
        "deprecated": {
          "type": "boolean"
        },
        "description": {
          "type": [
            "string",
            "null"
          ]
        },
        "else": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "enum": {
          "type": [
            "array",
            "null"
          ],
          "items": true
        },
        "exclusiveMaximum": {
          "type": [
            "number",
            "null"
          ],
          "format": "double"
        },
        "exclusiveMinimum": {
          "type": [
            "number",
            "null"
          ],
          "format": "double"
        },
        "format": {
          "type": [
            "string",
            "null"
          ]
        },
        "if": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "items": {
          "anyOf": [
            {
              "$ref": "#/definitions/SingleOrVec_for_Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "maxItems": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "maxLength": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "maxProperties": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "maximum": {
          "type": [
            "number",
            "null"
          ],
          "format": "double"
        },
        "minItems": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "minLength": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "minProperties": {
          "type": [
            "integer",
            "null"
          ],
          "format": "uint32"
        },
        "minimum": {
          "type": [
            "number",
            "null"
          ],
          "format": "double"
        },
        "multipleOf": {
          "type": [
            "number",
            "null"
          ],
          "format": "double"
        },
        "not": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "oneOf": {
          "type": [
            "array",
            "null"
          ],
          "items": {
            "$ref": "#/definitions/Schema"
          }
        },
        "pattern": {
          "type": [
            "string",
            "null"
          ]
        },
        "patternProperties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/Schema"
          }
        },
        "properties": {
          "type": "object",
          "additionalProperties": {
            "$ref": "#/definitions/Schema"
          }
        },
        "propertyNames": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "readOnly": {
          "type": "boolean"
        },
        "required": {
          "type": "array",
          "items": {
            "type": "string"
          }
        },
        "then": {
          "anyOf": [
            {
              "$ref": "#/definitions/Schema"
            },
            {
              "type": "null"
            }
          ]
        },
        "title": {
          "type": [
            "string",
            "null"
          ]
        },
        "type": {
          "anyOf": [
            {
              "$ref": "#/definitions/SingleOrVec_for_InstanceType"
            },
            {
              "type": "null"
            }
          ]
        },
        "uniqueItems": {
          "type": [
            "boolean",
            "null"
          ]
        },
        "writeOnly": {
          "type": "boolean"
        }
      },
      "additionalProperties": true
    },
    "SingleOrVec_for_InstanceType": {
      "anyOf": [
        {
          "$ref": "#/definitions/InstanceType"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/definitions/InstanceType"
          }
        }
      ]
    },
    "SingleOrVec_for_Schema": {
      "anyOf": [
        {
          "$ref": "#/definitions/Schema"
        },
        {
          "type": "array",
          "items": {
            "$ref": "#/definitions/Schema"
          }
        }
      ]
    }
  }
}